server/src/actors/events.rs

```diff
@@ -873,6 +872,11 @@ impl Handler<HandleEvent> for EventManager {
             .timeout(self.config.event_buffer_expiry(), ProcessingError::Timeout)
             .map(|_, _, _| metric!(counter("event.accepted") += 1))
             .map_err(clone!(project, captured_events, |error, _, _| {
+                // Do not track outcomes or capture events for non-event envelopes (such as
+                // individual attachments)
+                if !is_event {
+                    return;
```

Comment:
This will be a massive issue in the store endpoint. Since we allow direct ingestion of envelopes now, we have no way of knowing whether an outcome should be produced or not. 

It’s fine to defer a solution in that case, however.

Reply:
I think that is the solution, no? Since store accepts envelopes, and since envelopes may not necessarily represent events, we need to look into the envelope to know whether an outcome should be sent.

And for unparseable envelopes we just need to make an assumption that might turn out to be wrong. If we were to add out-of-band information about whether the envelope is an event, we would undermine the point of envelopes containing all necessary information.

Reply:
You’re right. I wrote this comment half-way through review. Let’s track this in the other comment thread.